src/main/java/edu/iastate/ece/sd/sdmay2126/orchestration/Job.java

```diff
@@ -0,0 +1,62 @@
+package edu.iastate.ece.sd.sdmay2126.orchestration;
+
+import edu.iastate.ece.sd.sdmay2126.application.Application;
+import edu.iastate.ece.sd.sdmay2126.application.ApplicationOutput;
+import edu.iastate.ece.sd.sdmay2126.application.ApplicationParameters;
+
+/**
+ * Represents a unique execution of some {@link edu.iastate.ece.sd.sdmay2126.application.Application}.
+ */
+public class Job {
+
+    private ApplicationParameters parameters;
+    private JobCompleted completionCallback;
+    private JobCanceled cancellationCallback;
+    private ApplicationOutput output;
+    private JobResult result = JobResult.UNKNOWN;
+
+    public Job(ApplicationParameters parameters) {
+        this(parameters, null, null);
+    }
+
+    public Job(ApplicationParameters parameters, JobCompleted completionCallback, JobCanceled cancellationCallback) {
+        if (parameters == null)
```

Comment:
Note: Could start using Kotlin?

Reply:
I definitely prefer Kotlin's null-safety. I've changed this to use a `@Notnull` annotation instead of a check.